runtime/drivers/duckdb/model_executor_localfile_self.go

```diff
@@ -0,0 +1,156 @@
+	}
+	if err := outputProps.Validate(opts); err != nil {
+		return nil, fmt.Errorf("invalid output properties: %w", err)
+	}
+
+	usedModelName := false
+	if outputProps.Table == "" {
+		outputProps.Table = opts.ModelName
+		usedModelName = true
+	}
+
+	materialize := true
+	if outputProps.Materialize != nil {
+		materialize = *outputProps.Materialize
+	}
+
+	asView := !materialize
+	tableName := outputProps.Table
+
+	// Prepare for ingesting into the staging view/table.
+	// NOTE: This intentionally drops the end table if not staging changes.
+	stagingTableName := tableName
+	if opts.Env.StageChanges {
+		stagingTableName = stagingTableNameFor(tableName)
+	}
+	if t, err := e.c.InformationSchema().Lookup(ctx, "", "", stagingTableName); err == nil {
+		_ = e.c.DropTable(ctx, stagingTableName, t.View)
+	}
+
+	// get the local file path
+	localPaths, err := e.from.FilePaths(ctx, opts.InputProperties)
+	if err != nil {
+		return nil, err
+	}
+	if len(localPaths) == 0 {
+		return nil, fmt.Errorf("no files to ingest")
+	}
+
+	if inputProps.Format == "" {
+		inputProps.Format = fileutil.FullExt(localPaths[0])
+	}
+
+	from, err := sourceReader(localPaths, inputProps.Format, inputProps.DuckDB)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the table
+	err = e.c.CreateTableAsSelect(ctx, stagingTableName, asView, "SELECT * FROM "+from, nil)
+	if err != nil {
+		_ = e.c.DropTable(ctx, stagingTableName, asView)
+		return nil, fmt.Errorf("failed to create model: %w", err)
+	}
+
+	// Rename the staging table to the final table name
+	if stagingTableName != tableName {
+		err = olapForceRenameTable(ctx, e.c, stagingTableName, asView, tableName)
+		if err != nil {
+			return nil, fmt.Errorf("failed to rename staged model: %w", err)
+		}
+	}
+
+	// Build result props
+	resultProps := &ModelResultProperties{
+		Table:         tableName,
+		View:          asView,
+		UsedModelName: usedModelName,
+	}
+	if inputProps.InvalidateOnChange {
+		resultProps.LocalFilePathsToHashes = make(map[string]any, len(localPaths))
+		// Need to store file path and its hashe to invalidate on change
+		for _, p := range localPaths {
+			resultProps.LocalFilePathsToHashes[p], err = fileHash(p)
+			if err != nil {
+				return nil, fmt.Errorf("failed to hash local file: %w", err)
+			}
+		}
+	}
+	resultPropsMap := map[string]interface{}{}
+	err = mapstructure.WeakDecode(resultProps, &resultPropsMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to encode result properties: %w", err)
+	}
+
+	// Done
+	return &drivers.ModelResult{
+		Connector:  opts.OutputConnector,
+		Properties: resultPropsMap,
+		Table:      tableName,
+	}, nil
+}
+
+func fileHash(path string) (string, error) {
+	file, err := os.Open(path)
+	if err != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	hasher := md5.New()
+	if _, err := io.Copy(hasher, file); err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(hasher.Sum(nil)), nil
+}
```

Comment:
Wouldn't it be simpler to produce one hash for all the files? You can keep copying file contents into the `hasher`. It anyways needs to reingest all files if one file is changed.

Reply:
Not currently. 
If I change one file then only the model that depends on that file will be re-ingested.

Reply:
I only meant all the files used by the one model being executed here

Reply:
Ah okay I misunderstood it is in the executor. 
Yes it can be a single hash.